bcipy/tasks/rsvp/copy_phrase.py

```diff
@@ -44,368 +71,610 @@ class RSVPCopyPhraseTask(Task):
+                       target_letter=target_letter,
+                       current_text=current_text,
+                       target_text=self.copy_phrase,
+                       next_display_state=next_state)
+
+        if not self.fake:
+            latest_evidence = self.copy_phrase_task.conjugator.latest_evidence
+            data.evidences = {
+                name: evidence if name in evidence_types else []
+                for name, evidence in latest_evidence.items()
+            }
+            data.likelihood = list(self.copy_phrase_task.conjugator.likelihood)
+        return data
+
+    def exit_display(self):
+        """Close the UI and cleanup."""
+        # Update task state and reset the static
+        self.rsvp.update_task_state(text=self.spelled_text,
+                                    color_list=['white'])
+        # Say Goodbye!
+        self.rsvp.text = trial_complete_message(self.window, self.parameters)
+        self.rsvp.draw_static()
+        self.window.flip()
+
+        # Give the system time to process
+        self.wait()
+
+    def update_session_data(self,
+                            data: Inquiry,
+                            save: bool = True,
+                            decision_made: bool = False) -> None:
+        """Update the current session with the latest inquiry data
+
+        Parameters
+        ----------
+        - data : inquiry to append to the session
+        - save : if True, persists the session to disk.
+
+        Modifies
+        --------
+        - self.session
+        """
+        self.session.add_sequence(data)
+        self.session.total_time_spent = self.experiment_clock.getTime()
+        if save:
+            self.write_session_data()
+        if decision_made:
+            self.session.add_series()
+
+    def write_session_data(self) -> None:
+        """Save session data to disk."""
+        if self.session:
+            _save_session_related_data(self.session_save_location,
+                                       self.session.as_dict())
+
+    def write_offset_trigger(self, trigger_file: TextIO):
+        """Append the offset to the end of the triggers file.
+
+        Parameters
+        -----------
+        - trigger_file : open file in which to write
+        """
+        if self.daq.is_calibrated:
+            _write_triggers_from_inquiry_copy_phrase(
+                ['offset', self.daq.offset],
+                trigger_file,
+                self.copy_phrase,
+                self.spelled_text,
+                offset=True)
+
+    def write_trigger_data(self, stim_times: List[Tuple[str, float]],
+                           trigger_file: TextIO) -> None:
+        """Save trigger data to disk.
+
+        Parameters
+        ----------
+        - stim_times : list of (stim, clock_time) tuples
+        - trigger_file : data will be appended to this file
+        """
+        # TODO: fix the helper to correctly handle the targetness of inquiry_preview stims.
```

Comment:
What does it write currently?

Reply:
I created a ticket for this with example output: https://www.pivotaltracker.com/story/show/178253198